app/models/policy_host.rb

```diff
@@ -8,10 +8,24 @@ class PolicyHost < ApplicationRecord
   validates :policy, presence: true
   validates :host, presence: true, on: :create
   validates :host_id, presence: true, uniqueness: { scope: :policy }
+  validate :host_supported?
+
+  scope :supported_versions, lambda { |policy_id|
+    profile = Profile.find_by(policy_id: policy_id)
+    profile&.supported_os_versions&.map(&:version)
```

Comment:
Be careful about `supported_os_versions` it's heavily inefficient if queried directly on profiles as it will produce a cache miss [here](https://github.com/RedHatInsights/compliance-backend/blob/master/app/models/profile.rb#L87). The method was optimized for calling it from GraphQL [underneath](https://github.com/RedHatInsights/compliance-backend/blob/master/app/models/os_major_version.rb#L56) the `OsMajorVersion` pseudo-model.

You can either recreate somehow that query optimized in your context, or come up with something else to make it fast.